extensions/vertx/deployment/src/main/java/io/quarkus/vertx/core/deployment/VertxCoreProcessor.java

```diff
@@ -286,6 +287,12 @@ ContextHandlerBuildItem createVertxContextHandlers(VertxCoreRecorder recorder, V
         return new ContextHandlerBuildItem(recorder.executionContextHandler(buildConfig.customizeArcContext()));
     }
 
+    @BuildStep(onlyIfNot = NativeOrNativeSourcesBuild.class)
```

Comment:
Does it make sense to do that outside of dev mode?

Reply:
It's really only for dev-mode, so yeah, we could limit it to that

Reply:
Yeah I would rather limit it to where it's useful. Either dev mode or dev/test mode, whatever is better.